persistence/pg.go

```diff
@@ -331,15 +362,66 @@ func (p *PostgresPersister) MarkHtlcSettled(ctx context.Context,
 			return types.ErrInvoiceNotFound
 		}
 
-		settledHash = &hash
+		invoiceHash = &invoice.Hash
 
 		return nil
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	return settledHash, nil
+	return invoiceHash, nil
+}
+
+func (p *PostgresPersister) selectInvoiceForUpdate(ctx context.Context,
```

Comment:
Refactor can go to a separate commit?

Reply:
Anything refactor can be pulled out of the PR and merged onto master, if it makes sense.